projects/packages/image-cdn/src/class-image-cdn.php

```diff
@@ -336,297 +367,342 @@ public static function parse_dimensions_from_filename( $src ) {
+				$source_type     = 'data-lazy-original';
+				$chosen_data_src = $data_lazy_original;
+			}
 
-					// Build URL, first maybe removing WP's resized string so we pass the original image to Photon.
-					if ( ! $fullsize_url && 0 === strpos( $src, $upload_dir['baseurl'] ) ) {
-						$src = self::strip_image_dimensions_maybe( $src );
-					}
+			// Update the src if one was provided in the `data-lazy-` attributes.
+			if ( 'src' !== $source_type ) {
+				$placeholder_src_orig = $src;
+				$placeholder_src      = $placeholder_src_orig;
+				$src_orig             = $chosen_data_src;
+				$src                  = $src_orig;
+			}
 
-					// Build array of Photon args and expose to filter before passing to Photon URL function.
-					$args = array();
+			// Check if image URL should be used with Photon.
+			if ( self::validate_image_url( $src ) ) {
+				$width  = $p->get_attribute( 'width' );
+				$height = $p->get_attribute( 'height' );
+
+				// First, check the image tag. Note we only check for pixel sizes now; HTML4 percentages have never been correctly
+				// supported, so we stopped pretending to support them in JP 9.1.0.
+				if ( ! is_string( $width ) || false !== strpos( $width, '%' ) ) {
+					$width = false;
+				}
 
-					if ( false !== $width && false !== $height ) {
-						$args[ $transform ] = $width . ',' . $height;
-					} elseif ( false !== $width ) {
-						$args['w'] = $width;
-					} elseif ( false !== $height ) {
-						$args['h'] = $height;
-					}
+				if ( ! is_string( $height ) || false !== strpos( $height, '%' ) ) {
+					$height = false;
+				}
+
+				// Detect WP registered image size from HTML class.
+				$class = $p->get_attribute( 'class' );
+				$size  = self::parse_class_name_prefixed_by( $class, 'size-' );
+				if ( null !== $size && false === $width && false === $height && 'full' !== $size && array_key_exists( $size, $image_sizes ) ) {
+					$width     = (int) $image_sizes[ $size ]['width'];
+					$height    = (int) $image_sizes[ $size ]['height'];
+					$transform = $image_sizes[ $size ]['crop'] ? 'resize' : 'fit';
+				} else {
+					unset( $size );
+				}
 
+				// WP Attachment ID, if uploaded to this site.
+				$attachment_id = self::parse_class_name_prefixed_by( $class, 'wp-image-' );
+				$images        = array();
+				$index         = 0;
+				if (
+					$attachment_id &&
+					preg_match( '#^[1-9][0-9]*$#', $attachment_id ) &&
+					0 === strpos( $src, $upload_dir['baseurl'] ) &&
 					/**
-					 * Filter the array of Photon arguments added to an image when it goes through Photon.
-					 * By default, only includes width and height values.
+					 * Filter whether an image using an attachment ID in its class has to be uploaded to the local site to go through Photon.
 					 *
-					 * @see https://developer.wordpress.com/docs/photon/api/
+					 * @TODO: What is the point of passing $images and $index. Are they required?
```

Comment:
I looked around but couldn't figure out where that filter is being used today. It was used in the past, but may not be used anymore today.